.NET/Microsoft.Recognizers.Text.DataTypes.DataDrivenTests/TestTimexResolver.cs

```diff
@@ -64,6 +64,177 @@ public void DataTypes_Resolver_Date_Sunday()
+            var resolution = TimexResolver.Resolve(new[] { "XXXX-02-02 " }, today);
+            Assert.AreEqual(2, resolution.Values.Count);
+
+            Assert.AreEqual("XXXX-02-02", resolution.Values[0].Timex);
+            Assert.AreEqual("date", resolution.Values[0].Type);
+            Assert.AreEqual("2020-02-02", resolution.Values[0].Value);
+            Assert.IsNull(resolution.Values[0].Start);
+            Assert.IsNull(resolution.Values[0].End);
+
+            Assert.AreEqual("XXXX-02-02", resolution.Values[1].Timex);
+            Assert.AreEqual("date", resolution.Values[1].Type);
+            Assert.AreEqual("2021-02-02", resolution.Values[1].Value);
+            Assert.IsNull(resolution.Values[1].Start);
+            Assert.IsNull(resolution.Values[1].End);
+        }
+
+        [TestMethod]
+        public void DataTypes_Resolver_DateTimeRange_Oct_25th_Afternoon()
+        {
+            var today = new System.DateTime(2020, 10, 20);
+            var resolution = TimexResolver.Resolve(new[] { "XXXX-10-25TAF" }, today);
+            Assert.AreEqual(2, resolution.Values.Count);
+
+            Assert.AreEqual("XXXX-10-25TAF", resolution.Values[0].Timex);
+            Assert.AreEqual("datetimerange", resolution.Values[0].Type);
+            Assert.AreEqual("2019-10-25 12:00:00", resolution.Values[0].Start);
+            Assert.AreEqual("2019-10-25 16:00:00", resolution.Values[0].End);
+            Assert.IsNull(resolution.Values[0].Value);
+
+            Assert.AreEqual("XXXX-10-25TAF", resolution.Values[1].Timex);
+            Assert.AreEqual("datetimerange", resolution.Values[1].Type);
+            Assert.AreEqual("2020-10-25 12:00:00", resolution.Values[1].Start);
+            Assert.AreEqual("2020-10-25 16:00:00", resolution.Values[1].End);
+            Assert.IsNull(resolution.Values[1].Value);
+        }
+
+        [TestMethod]
+        public void DataTypes_Resolver_DateTimeRange_Week11_Monday()
+        {
+            var today = new System.DateTime(2020, 10, 20);
+            var resolution = TimexResolver.Resolve(new[] { "XXXX-W11-1" }, today);
+            Assert.AreEqual(2, resolution.Values.Count);
+
+            Assert.AreEqual("XXXX-W11-1", resolution.Values[0].Timex);
+            Assert.AreEqual("date", resolution.Values[0].Type);
+            Assert.IsNull(resolution.Values[0].Start);
+            Assert.IsNull(resolution.Values[0].End);
+            Assert.AreEqual("2020-03-09", resolution.Values[0].Value);
+
+            Assert.AreEqual("XXXX-W11-1", resolution.Values[1].Timex);
+            Assert.AreEqual("date", resolution.Values[1].Type);
+            Assert.IsNull(resolution.Values[1].Start);
+            Assert.IsNull(resolution.Values[1].End);
+            Assert.AreEqual("2021-03-15", resolution.Values[1].Value);
+        }
+
+        [TestMethod]
+        public void DataTypes_Resolver_DateTimeRange_Thanksgiving()
+        {
+            var today = new System.DateTime(2020, 10, 20);
```

Comment:
Could you add comments in the less straightforward test cases like this?
// XXXX-11-WXX-4-4 -> 4th Thursday (4th ISO weekday) in unspecified week in November in unspecified year
This will help future folks seeing this to understanding both the timexes and the tests themselves.